Scripts/Game/ODIN/Editor/Containers/Actions/ContextActions/ODIN_TeleportToSquadmateContextAction.c

```diff
@@ -0,0 +1,134 @@
+//------------------------------------------------------------------------------------------------
+[BaseContainerProps(), SCR_BaseContainerCustomTitleUIInfo("m_Info")]
+class ODIN_TeleportToSquadmateContextAction: SCR_BaseContextAction
+{	
+	override bool CanBeShown(SCR_EditableEntityComponent hoveredEntity, notnull set<SCR_EditableEntityComponent> selectedEntities, vector cursorWorldPosition, int flags)
+	{
+		if (!hoveredEntity)
+			return false;
+		
+		GenericEntity owner = hoveredEntity.GetOwner();
+		if (!owner) 
+			return false;
+		
+		if (!SCR_ChimeraCharacter.Cast(owner))
+			return false;
+		
+		// only able to do on players
+		if (!EntityUtils.IsPlayer(owner))
+			return false;
+		
+		return true;
+	}
+	
+	override bool CanBePerformed(SCR_EditableEntityComponent hoveredEntity, notnull set<SCR_EditableEntityComponent> selectedEntities, vector cursorWorldPosition, int flags)
+	{
+		return CanBeShown(hoveredEntity, selectedEntities, cursorWorldPosition, flags);
+	}
+	
+	override void Perform(SCR_EditableEntityComponent hoveredEntity, notnull set<SCR_EditableEntityComponent> selectedEntities, vector cursorWorldPosition,int flags, int param = -1)
+	{
+		if (!hoveredEntity)
+			return;
+		
+		GenericEntity owner = hoveredEntity.GetOwner();
+		
+		MenuBase myMenu = GetGame().GetMenuManager().OpenDialog(ChimeraMenuPreset.ODIN_ListBox_ID, DialogPriority.INFORMATIVE, 0, true); 
+		
+		// null check 
+		if (!myMenu)
+		{
+			ODIN_LogHelper.Log("Could not find listbox dialog defined", "TeleportToSquadmateContextAction", LogLevel.WARNING);
+			return;	
+		}
+		
+		// cast so we have dialog object. Set hovered-entity, we need to get squadmates from it and have it to teleport it
+		ODIN_ListboxSingleSelectDialog myMenuUI = ODIN_ListboxSingleSelectDialog.Cast(myMenu);
+		myMenuUI.SetHoveredEntity(hoveredEntity);
+		
+		// set Title 
+		myMenuUI.SetTitle("#ODIN-ActionList_TeleportToSquadmate_Name");
+			
+		// Get squadmates 
+		SCR_AIGroup group = ODIN_GroupHelper.GetGroup(owner);
+		if (!group || group.GetPlayerCount() <= 0) 
+		{
+			// TODO give error message back to zeus that unit is not in a group 
+			ODIN_LogHelper.Log("null group or <= 0 in group", "TeleportToSquadmateContextAction", LogLevel.WARNING);
+			return;
+		}
+		
+		// can't get character entities directly from group, only for AI. So for players we get their IDs and gotta find and cast them with that
+		array<int> members = group.GetPlayerIDs();
+		foreach (int playerID : members)
+		{
+			PlayerManager playerManager = GetGame().GetPlayerManager();
+			string name = playerManager.GetPlayerName(playerID);
+			ChimeraCharacter player = ChimeraCharacter.Cast(playerManager.GetPlayerControlledEntity(playerID));
+			
+			// if player is dead and the object is null, we skip
+			if (!player)
+				continue;
+			
+			// don't add the unit we want to teleport
+			if (player.GetID() == owner.GetID())
```

Comment:
I think comparing `player` with `owner` directly would also work

Reply:
Just checked. That seems to work. Was unsure if the object comparison was implemented.